src/traces/bar/set_positions.js

```diff
@@ -633,14 +638,21 @@ function normalizeBars(gd, sa, sieve) {
     }
 
     // update range of size axis
-    Axes.expand(sa, sRange, {tozero: true, padded: padded});
+    var extremes = Axes.findExtremes(sa, sRange, {tozero: true, padded: padded});
+    putExtremes(traces, sa, extremes);
 }
 
 
 function getAxisLetter(ax) {
     return ax._id.charAt(0);
 }
 
+function putExtremes(cd, ax, extremes) {
+    for(var i = 0; i < cd.length; i++) {
+        cd[i][0].trace._extremes[ax._id] = extremes;
+    }
+}
```

Comment:
Now that we're not trying to make `_extremes` survive visibility changes, does it actually help to have the same thing repeated in all traces here? Seems like that will just waste time later on. Can we just put it in the first one and clear all the others?

Reply:
Thanks for pointing this out, but turns out we still need to attach `_extremes` to all bar traces at the moment for error bars. Error bars concat their min/max extremes into the `trace._extremes` in which they are linked and assume that `trace._extremes[ax._id]` exists.

We could alternatively linked error bar `_extremes` in their `trace.error_(x|y)` containers and make `getAutorange` look for `error_(x|y)` in each trace. Or perhaps, making error bar create `trace._extremes[ax._id]` containers when needed would suffice.

But, I wouldn't worry too much about performance. I can't imagine `getAutorange` taking more than 1ms even in the worse of scenario.

Reply:
Ah OK. I suppose the first one could have the actual values, then all the others could just get `[]` to give something for errorbars to append to... but yeah, no big deal, no block. These extras will get pruned quickly during your improved `_concat` that I hadn't noticed when I made that comment.